tests/form-filling.js

```diff
@@ -0,0 +1,76 @@
+const helper = require('./helper');
+const genericFormPo = require('./page-objects/forms/generic-form.po');
+const answeredQuestions = [];
+const flatten = require('flat');
+
+const checkbox = (answer, answerVal) => {
+  const checks = answerVal.split();
+  checks.forEach(async (check) => {
+    await helper.clickElementNative(element(by.css(`${answer.css}[value=${check}]`)));
+  });
+};
+
+const radio = async (answer, answerVal) => {
+  await helper.clickElementNative(element(by.css(`${answer.css}[value="${answerVal}"]`)));
+};
+
+const date = async (answer, answerVal) => {
+  const elm = element(by.css(`${answer.css} + div input`));
+  await helper.waitUntilReadyNative(elm);
+  await elm.sendKeys(answerVal);
+};
+
+const defaultAction = async (answer, answerVal) => {
+  const elm = element(by.css(answer.css));
+  await helper.waitUntilReadyNative(elm);
+  await elm.sendKeys(answerVal);
+};
+
+const answerActions = { checkbox, radio, date };
+
+const repeats = (pathKeys) => {
+  const newKeys = pathKeys;
```

Comment:
I don't think this is necessary?

Reply:
This is handling when repeats in forms are used. The flat library treats them as an array, as it should, but that causes a disconnect between the css path `/delivery/baby_repeats/name` and the report `delivery.baby_repeats.0.name`. I took this route for now because I'm not sure how much our forms use repeats but in this case it's only 1 time. When we get to needing a form and using lots of repeats I think it would be good to tackle then.

Reply:
Ahh got it. Thanks.